fuzz/fuzz_targets/buf_independent.rs

```diff
@@ -75,49 +75,65 @@ mod intermittent_eofs {
         pub fn new(inner: R) -> Self {
             Self {
                 inner,
-                enabler: Rc::new(Enabler::new()),
+                controller: Rc::new(EofController::new()),
                 eof_soon: true,
             }
         }
 
-        pub fn enabler(&self) -> Rc<Enabler> {
-            self.enabler.clone()
+        pub fn controller(&self) -> Rc<EofController> {
+            self.controller.clone()
         }
     }
 
     impl<R: Read> Read for IntermittentEofs<R> {
         fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
-            if self.enabler.is_enabled() && self.eof_soon {
+            if self.controller.are_intermittent_eofs_enabled() && self.eof_soon {
                 self.eof_soon = false;
                 return Ok(0);
             }
 
             self.eof_soon = true;
-            self.inner.read(buf)
+            let inner_result = self.inner.read(buf);
+            match inner_result.as_ref() {
+                Ok(0) => self.controller.mark_inner_eof(),
+                _ => (),
+            }
+
+            inner_result
         }
     }
 
-    pub struct Enabler {
-        is_enabled: RefCell<bool>,
+    pub struct EofController {
+        are_intermittent_eofs_enabled: RefCell<bool>,
```

Comment:
For copyable types `Cell<bool>` works as well, and has zero overhead

Reply:
Thanks for the feedback - I didn't realize this was an option.  I like the resulting simplification.  Done.